crates/bevy_ecs/src/query/mod.rs

```diff
@@ -21,6 +21,12 @@ pub use par_iter::*;
 pub use state::*;
 pub use world_query::*;
 
+use crate as bevy_ecs;
+
+/// A marker component that excludes an entity from queries that don't specifically request them.
+#[derive(bevy_ecs_macros::Component)]
```

Comment:
This type needs to be `Reflect`, `Clone` and `Copy`, and the type should be registered in `bevy_core`. This is needed for various serialization use cases, namely networking and save games.

Reply:
The component should probably also use sparse storage to avoid table moves.

Reply:
Hmm. It would slow down the iteration in turn though. I think we stick to the simplest design for now until we can benchmark it.

Reply:
It doesn't slow down iteration unless you are accessing the component. Since it's a ZST that will likely be never.

Reply:
Oh actually I suppose you are correct since most queries now need to filter on this, disregard.